Completely rework EventT

ch3pjw opened the pull request:
to allow for querying state and to allow for writing emitted events as we
go, rather than collecting them all up at the end. This makes failure
behaviour better/more intuitive.

foolswood commented on src/Registration.hs:
```diff
@@ -62,7 +63,8 @@ type Salt = Text
-  secondsToDiffTime $ 60 * 60 * 24
+  -- secondsToDiffTime $ 60 * 60 * 24
+  secondsToDiffTime $ 6
 
```
Erk!

ch3pjw replied:
Woops. Force pushed a fix.

foolswood commented on src/Lib.hs:
```diff
@@ -183,6 +183,8 @@ interestedResource actor name req sendResponse =
         present <- hasEmail uuid
         if present
         then do
+          -- FIXME: this now comes back with a result saying if you've
```
The idea being to come back and give a bit more feedback to the user here?

ch3pjw replied:
Yeah, that's right. Didn't want to make a behaviour change here, so left it like it was for now.

foolswood reviewed (changes requested):
Pretty much just that timeout from what I can grok.

ch3pjw committed:
Completely rework EventT

to allow for querying state and to allow for writing emitted events as we
go, rather than collecting them all up at the end. This makes failure
behaviour better/more intuitive.
